### PR TITLE
CI: run helix, and non-helix tests in separate jobs

### PR DESCRIPTION
**Pull request by radical**
[ci] Run non-helix, and helix tests in separate jobs

## Public pipeline

These will the build jobs now on the public pipeline:

- `Windows`
    - runs non-helix tests
    - and package signing, and workload build
- `Windows_helix_tests`
    - runs helix tests
- `Linux`
    - runs non-helix tests
- `Linux_helix_tests`
    - runs helix tests

Code-coverage is unaffected, and runs after all the build jobs are done.

## Internal pipeline

The internal pipeline is effectively unchanged, running both helix, and
non-helix tests on the Windows job.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5069)

### CODE REVIEW

File: eng/pipelines/templates/BuildAndTest.yml [hunk 1]
```diff
@@ -86,29 +87,30 @@ steps:
...
+  - ${{ if ne(parameters.runAsPublic, 'true') }}:
+    - task: 1ES.PublishPipelineArtifact@1
+      displayName: Publish code coverage results
+      inputs:
+        targetPath: '${{ parameters.repoArtifactsPath }}/CodeCoverage'
+        artifactName: '$(Agent.JobName)_CodeCoverageResults'
 
-  - ${{ if eq(parameters.isWindows, 'true') }}:
+  # Run on windows, for internal pipeline, or public+non-helix-tests job
+  - ${{ if and(eq(parameters.isWindows, 'true'), or(ne(parameters.runAsPublic, 'true'), ne(parameters.runHelixTests, 'true'))) }}:
```
Comment (eerhardt) [inline eng/pipelines/templates/BuildAndTest.yml:113]: Is there a way to make this a bit more readable? It is really hard to parse/understand for me.

**Review by eerhardt — APPROVED** [review]
LGTM

Thanks!

File: eng/pipelines/templates/BuildAndTest.yml (hunk 1 above)
Comment (radical) [reply eng/pipelines/templates/BuildAndTest.yml:113]:
```yml
  # Run on windows, for internal pipeline, or public+non-helix-tests job
  - ? ${{ if and(
               eq(parameters.isWindows, 'true'),
               or(
                  ne(parameters.runAsPublic, 'true'),
                  ne(parameters.runHelixTests, 'true'))) }}
    : # on a new line as the condition above is flattened into a single line
    - script: ${{ parameters.buildScript }}
              -pack
              -sign $(_SignArgs)
              -publish $(_PublishArgs)
              -configuration ${{ parameters.buildConfig }}
              /bl:${{ parameters.repoLogPath }}/pack.binlog
              /p:Restore=false /p:Build=false
              $(_OfficialBuildIdArgs)
      displayName: Pack, Sign, and Publish
```

I'm not sure how much that helps readability. I can't add comments inline because that would be part of the expression then. I could add a variables section to name parts of this like `isInternalOrNonHelixTestsJob`?

Comment (radical) [reply eng/pipelines/templates/BuildAndTest.yml:113]: Merging the PR, but I'll open a new one for any suggested changes from this.
